### PR TITLE
Update lease timing to reflect current round deadlines

### PR DESCRIPTION
**Pull request by santhnm2**

### CODE REVIEW

File: scheduler/scheduler.py [hunk 1]
```diff
@@ -2666,24 +2669,34 @@ def _update_lease_callback(self, job_id, worker_id, steps, duration,
                                                         max_steps,
                                                         max_duration))
 
-        # Round the remaining steps to the nearest multiple of scale_factor.
-        scale_factor = self._jobs[job_id].scale_factor
-        remaining_steps = self._get_remaining_steps(job_id)
-        remaining_steps = int(math.ceil(remaining_steps / scale_factor))
+            # Round the remaining steps to the nearest multiple of scale_factor.
```
Comment (deepakn94) [inline scheduler/scheduler.py:2672]: You want to access these variables with the lock held, correct?

File: scheduler/scheduler.py [hunk 2]
```diff
@@ -2666,24 +2669,34 @@ def _update_lease_callback(self, job_id, worker_id, steps, duration,
...
+            return (remaining_steps, remaining_time_in_current_round)
 
         # Extend the lease if the job has been placed on the same workers
         # for the upcoming round.
         with self._scheduler_lock:
             # TODO: Remove scan of self._jobs_with_extended_lease.
             for job_id_combination in self._jobs_with_extended_lease:
                 if job_id.overlaps_with(job_id_combination):
-                    return (max_steps, max_duration + self._time_per_iteration)
+                    updated_lease_duration = max_duration
```
Comment (deepakn94) [inline scheduler/scheduler.py:2693]: What is `max_duration`? Maximum duration across tasks?

Comment (santhnm2) [reply scheduler/scheduler.py:2693]: `max_duration` is the lease duration

File: scheduler/scheduler.py (hunk 1 above)
Comment (santhnm2) [reply scheduler/scheduler.py:2672]: Yes

File: scheduler/scheduler.py [hunk 3]
```diff
@@ -2643,17 +2643,20 @@ def _init_job_callback(self, job_id):
...
-            extra_time = 0
             if (self._next_worker_assignments is not None and
                 next_job_combination is not None):
-                current_time = self.get_current_timestamp()
-                elapsed_time_in_round = \
-                    current_time - self._current_round_start_time
-                extra_time += self._time_per_iteration - elapsed_time_in_round
-                extra_time = max(extra_time, 0)
-            return (remaining_steps, self._time_per_iteration, extra_time)
+                extra_time = remaining_time_in_current_round
```
Comment (deepakn94) [inline scheduler/scheduler.py:2655]: Why do you need this variable?

Comment (santhnm2) [reply scheduler/scheduler.py:2655]: I don't, it was left over from the previous version - will remove

File: scheduler/scheduler.py [hunk 4]
```diff
@@ -2643,17 +2643,20 @@ def _init_job_callback(self, job_id):
...
+                return (remaining_steps, self._time_per_iteration,
```
Comment (deepakn94) [inline scheduler/scheduler.py:2656]: Add a comment somewhere what the tuple you return means

Comment (santhnm2) [reply scheduler/scheduler.py:2658]: Done

File: scheduler/scheduler.py (hunk 3 above)
Comment (santhnm2) [reply scheduler/scheduler.py]: Done

**Pull request merged**